inspirehep/modules/workflows/tasks/merging.py

```diff
@@ -50,7 +53,7 @@ def get_head_source(head_uuid):
         * None if there are no root records
     """
     roots_sources = set(
-        r.source for r in
+        r.source.name for r in
         WorkflowsRecordSources.query.filter_by(record_id=head_uuid).all()
     )
 
```

Comment:
@michamos shall we fix the logic here or leave it as it is?

Reply:
what do you mean?

Reply:
right, this is used by the manual merger only to decide which strategy to use in that case. I would say the docstring is right and the behaviour is wrong, as we are always saying that publisher is stronger than arXiv because we don't want to overwrite the work of the publisher by the arXiv version. However, this doesn't account for weird cases where you want to merge stuff from several different publishers and has never been really tested. We need to revisit this down the line when we work on manual merges.

Reply:
Then we will leave it like this for now :)

Reply:
That's not what I said. You need to change it to be the same as the docstring, but I am saying that it's not 100% sure that this is the right behaviour anyway.

Reply:
As we discussed IRL we can drop this function and remove it from the ``manual merger`` ([here](https://github.com/inspirehep/inspire-next/blob/8d9e7dc234ee64653113b9c001c02df1985668db/inspirehep/modules/workflows/workflows/manual_merge.py#L83) and [here](https://github.com/inspirehep/inspire-next/blob/4a123080ac1d2cdd6f445a3104f8ef06b99edb44/inspirehep/modules/workflows/tasks/manual_merging.py#L59))